ts/modules/accessibility/focusBorder.ts

```diff
@@ -73,9 +73,27 @@ extend(H.SVGElement.prototype, {
         bb.x += this.translateX ? this.translateX : 0;
         bb.y += this.translateY ? this.translateY : 0;
 
+        let borderPosX: number,
+            borderPosY: number;
+
+        // For text elements, apply x and y offset, #11397.
+        if (this.element.nodeName === 'text') {
+            const isRotated = Boolean(this.rotation);
+            const isFirefox = H.isFirefox;
+            borderPosX = this.attr('x') as number - (bb.width * 0.5) - pad +
+                // Correct the offset caused by the browser.
+                (isRotated ? bb.height * 0.068 : 0);
+            borderPosY = this.attr('y') as number - (bb.height * 0.5) - pad +
+                // Firefox needs different correction value.
+                (isRotated ? 0 : -bb.height * (isFirefox ? 0.25 : 0.068));
+        } else {
+            borderPosX = bb.x - pad;
+            borderPosY = bb.y - pad;
```

Comment:
If you will set those values as default ones in lines 76-77 then the type of the variables will be closer to the set definitions (`number`) - now they start as `undefined`.